src/cpp/thread_manager/thread_manager.h

```diff
@@ -144,8 +144,7 @@ class ThreadManager {
       thread_creator_;
   std::function<void(gpr_thd_id)> thread_joiner_;
 
-  std::mutex list_mu_;
```

Comment:
Could you clarify why mutex is removed?  (I see that you are using `mu_` for list operations as well - I am ok with this change but I want to know if you did it for simplifying code (and perhaps reducing mutex acquisitions) or was this actually redundant ?

Reply:
My main goal was to avoid error-proneness; having 2 locks in the same object makes it more confusing, IMO, as to when the object is really locked. I would prefer having a coarse-grain lock unless it is a provable negative impact, and given the scope of the list_mu operations, I just didn't see that being likely to be substantial.